crates/uv-pypi-types/src/conflicts.rs

```diff
@@ -88,6 +246,42 @@ impl ConflictSet {
         self.iter()
             .any(|set| set.package() == package && *set.conflict() == conflict)
     }
+
+    /// Returns true if these conflicts contain any set that contains the given
+    /// [`ConflictItem`].
+    pub fn contains_item(&self, conflict_item: &ConflictItem) -> bool {
+        self.set.contains(conflict_item)
+    }
+
+    /// This [`ConflictSet`] was inferred from directly defined conflicts.
+    pub fn is_inferred_conflict(&self) -> bool {
+        self.is_inferred_conflict
+    }
+
+    // FIXME: Error if old is not present
+    /// Replace an old [`ConflictItem`] with a new one.
+    pub fn replaced_item(
+        &self,
+        old: &ConflictItem,
+        new: ConflictItem,
+    ) -> Result<Self, ConflictError> {
+        let mut new_set = self.set.clone();
+        if !new_set.contains(old) {
+            return Err(ConflictError::ReplaceMissingConflictItem);
+        }
+        new_set.remove(old);
+        new_set.insert(new);
+        Ok(Self {
+            set: new_set,
+            is_inferred_conflict: false,
+        })
+    }
+
+    /// Mark this [`ConflictSet`] as being inferred from directly
+    /// defined conflicts.
+    pub fn set_as_inferred_conflict(&mut self) {
```

Comment:
Does this need to be public?

Reply:
Nope, it's being called from a newtype. Updated